### ISSUE TITLE
 Blob lieutenant: "Shadow" #66869 Is confusing, shouldn't spawn entities inside a base, and spawns too many minions which then persist

### ISSUE THREAD

**Issue by Voideka** [CleverRaven/Cataclysm-DDA#69673]
### Describe the bug

Bullet points:

- Shadow event started with me standing 2 tiles from the hunting lodge I use as a base, which according to the commit should not happen ("They specifically wait to attack you until you're far from shelter").

- Not a bug per se, but the initial event text is confusing and did not clearly communicate to me that something bad was happening.

- Swarming Amalgamations can self-revive, which completely bypasses the lifespan restriction of the spawned enemies, and the limit of 6 at a time.

-------------------------

I was standing outside my hunting lodge disassembling a fridge at night, when the Shadow event started. The text "You shiver as the night closes in" initially left me confused, and made me think it was just telling me my character was too cold, so I continued working on the fridge. Soon afterwards I was interrupted by several Swarming Amalgams, coming from inside my base. Having no idea what was going on, I googled, and found a reddit thread with another confused person, who was told to use a flashlight to ward the event off.

I then activated my flashlight and ran around as the Swarming Amalgamations seemingly died from exposure to the light. After a bit the event seemed to end, and I went around pulping the corpses and repairing the damage done to my base, as the sun rose. I then discovered several still living Swarming Amalgamations in my base, and also several running around outside that did not die when the event ended. I tried killing them with a flashlight, but that did not work. My character is not very combat focused, so I will probably have to abandon my base at this point.

### Attach save file

[Lilymoor-trimmed.tar.gz](https://github.com/CleverRaven/Cataclysm-DDA/files/13447196/Lilymoor-trimmed.tar.gz)

### Steps to reproduce

Hang out near a base in a forest until the Shadow event triggers. Fail to find some of the dozens of spawned enemies.

### Expected behavior

Warning prompt should be clearer. Something akin to "Night closes in on you, and something feels wrong." would be way less confusing, without giving away exactly what's happening.

Shadow event should not occur near shelter, and it should certainly not spawn stuff inside of it.

Spawned enemies should fully despawn and not leave behind self-reviving corpses. These things are all over the place, some hiding in the forest, some on/in the river next to it, and it is simply unreasonable to expect me to be able to find and pulp all of them.

### Versions and configuration

- OS: Windows
    - OS Version: 10.0.19045.3693 (22H2)
- Game Version: 75cf733 [64-bit]
- Graphics Version: Tiles
- Game Language: System language []
- Mods loaded: [
    Dark Days Ahead [dda],
    Disable NPC Needs [no_npc_food],
    Portal Storms Ignore NPCs [personal_portal_storms],
    Slowdown Fungal Growth [no_fungal_growth],
    sees-player, run, crouch, bleed icons, retrodays [sees_player_retro]
]

**Comment by RenechCDDA** [comment]
I've changed the tags on your issue from being a bug report to being a feature request/suggestion/discussion, as that what is what it appears to be.

>Shadow event started with me standing 2 tiles from the hunting lodge I use as a base, which according to the commit should not happen ("They specifically wait to attack you until you're far from shelter").

To be explicitly clear, that is qualified by the statement "(as far as my dumb logic can check, anyway)". What this means that it is checking against the OMT you are on. There is no reasonable way to check all nearby OMTs for any possible shelter. This works in the player's favor, as you can be only a short distance from cover and thus have a very easy escape.

>Swarming Amalgamations can self-revive, which completely bypasses the lifespan restriction of the spawned enemies, and the limit of 6 at a time.

The time for them to revive after expiring should be the whatever the normal time is, several hours or so. I do not foresee any encounters that should last that long - that is enough time for you to walk many many many OMTs distance away and escape.

You may be confused by the (intentional) fact that they have varied lifespans of 15-30 seconds, while the 'spell' that summons them has an effective cooldown of 20 seconds.

I'd like to note that changing the monsters to not have corpses would be simple-ish, the preferred way to go about it would probably be to make copies of the monster which use copy-from and only define a single new value - `"corpse_type": "NO_CORPSE"`.

**Comment by GuardianDll** [comment]
> changing the monsters to not have corpses

i suppose adding `no_corpse` boolean for `u_spawn_monster` would be much robust solution, or rearranging EoC to use spell, that already has `NO_CORPSE` flag (that iirc doesn't work currently, but should)

**Comment by GuardianDll** [comment]
do monstergroups support adding custom flags to monsters in groups, similarly to item groups?
```json
{ "item": "cotton_patchwork", "prob": 30, "custom-flags": [ "FILTHY" ] },
```

**Comment by Voideka** [comment]
> To be explicitly clear, that is qualified by the statement "(as far as my dumb logic can check, anyway)". What this means that it is checking against the OMT you are on. There is no reasonable way to check all nearby OMTs for any possible shelter. This works in the player's favor, as you can be only a short distance from cover and thus have a very easy escape.

I had a feeling that was the case, but to say it works in the player's favor seems a little odd when the basis of this report is a situation that does not work in my favor: my base was overrun by enemies I cannot and was not prepared to deal with, as this was my first encounter with any of these mechanics/mobs.


> You may be confused by the (intentional) fact that they have varied lifespans of 15-30 seconds, while the 'spell' that summons them has an effective cooldown of 20 seconds.

I was taking the comments in the JSON at face value: "The maximum lifespan should not be terribly longer than the cooldown(currently 20 seconds), or else multiple groups can be active at once and make this much harder than desired."

It seemed to me that you were trying to prevent the encounter from exceeding a certain difficulty, and IMO having random corpses strewn about my base that I have no realistic way to deal with before they revive is contrary to the intended difficulty. Again, corpses were in the middle of a river, and hidden amongst a relatively dense forest. To expect the player to be able to find every last one of the corpses that resulted from the encounter is unrealistic, and incredibly punishing when they inevitably are unable to do so. I pulped somewhere around a dozen corpses, and there are still at least half a dozen Swarming Amalgams in and around my base.

> 
> I'd like to note that changing the monsters to not have corpses would be simple-ish, the preferred way to go about it would probably be to make copies of the monster which use copy-from and only define a single new value - `"corpse_type": "NO_CORPSE"`.

This would or removing the REVIVE S tag would be my suggested change, along with a slight clarification in the initial event text to better clue the player in to what's happening.

In all honesty, while I was completely caught off guard and confused by the event, I found it enjoyable. It was a nice change of pace that involved actual interaction and problem-solving (though that was dampened a bit by the googling). It was the resultant re-spawning of the monsters that make things feel overbearing and unfun.

**Comment by Spicyshadow** [comment]
It feels wierd that the somewhat supernatural incoporial shadow leutanant spawns physical creatures like amalgamations at all. It would be much more appropriate if it spawned a shadow version of the swarmer amalgamation that melted away when it died just like the leutanant itself. Having it generate a heap of reviving corpses doesn´t add much besides being hugely annoying to the player.

**Comment by CoroNaut** [comment]
The event should probably get a re-work.

1. The shadow creature is sitting in direct light from a floodlight with no issue. It doesn't try to run away from the light or is even damaged by it. 
2. The event spawns just so many amalgamations and spawns so many corpses. It will absolutely demolish anyone ill-prepared for it. 
3. It spawned amalgamations not even 5 tiles from the player sometimes.
4. The brutish amalgamations it spawns can't see the player in any reasonable way, from [Hitchhikers guide](https://cdda-guide.nornagon.net/monster/mon_amalgamation_soldier), it says vision range is 20 (day) / 8 (night). It is clearly extremely bright despite it being night, so it should probably see the player.
5. If the amalgamations do spawn far from the player with the floodlight, the amalgamations just end up dying before they even reach the player. IF they even see the player at the distance they are spawned at
6. If you are already too prepared, it should probably spawn more harder enemies.
7. Honestly the corpses are just too much, I mean there is 300 already at the time of writing.

Some expectations of things could go:
1. I probably expect that the shadow just up an poofs if its in bright light for too long. 
2. Have the event scale (if it doesn't already) faster
3. The amalgamations could probably spawn specifically out of the players line of sight at all times. If there is a massive amount of light, this would be behind trees or buildings, or the shadow could create black mist or something for them
4. Allow the brutish amalgamations their 20 vision range if there is _any_ amount of bright light, like a floodlight.
5. Have the amalgamations attack (or attempt to) the player at least once or twice before they just up and die. Also perhaps allow them some form of clairvoyance if they are too far.
6. Have the event scale faster for players (perhaps depending on the value of their worn gear at the time).
7. Corpses should probably disappear, or perhaps if you try to interact with their corpses you get a message "The corpse suddenly vanishes" or something similar.


First image showing that some amalgamations can't even see me, and the second showing just how many have spawned
![image](https://github.com/CleverRaven/Cataclysm-DDA/assets/24307650/989032a0-82cc-42cd-9160-e597b9272b0e)
![image](https://github.com/CleverRaven/Cataclysm-DDA/assets/24307650/c4b1c954-b4e7-48fb-b649-2cb5bb32fb5b)

Don't ask why the chicken corpse has a hoe inside of it lol.

**Comment by Voideka** [comment]
Hopefully my commenting on my own issue isn't violating some git-hub etiquette thing, but I've been thinking about this particular event a fair bit and wanted to at least finalize my thoughts here.

As I see it there are several issues with this event as it stands:

1. The event literally cannot function as intended, as the EoC system currently has no way to detect shelter.

The solution here is either to extend the EoC system so that it can detect shelter, for example by adding a terrain_flag option to u_location_variable (if that's possible), or to rethink the event with the understanding that it will be spawning on/near people's bases. Setting up shelter near forests and fields is relatively common in my experience, meaning there is a high likelihood the event will spawn on a player's base.

2. The Shadow Lieutenant can either spawn indoors, inside structures, or too easily destroys structures. After the event that prompted this submission occurred, I discovered one of the walls leading to a bedroom in my hunting lodge had been destroyed. There were no amalgams in that area, so it was something the Shadow had to have done.

This should not occur. There needs to be some check to make sure they aren't spawning indoors. If instead what happened is that the Shadow demolished the wall, I feel as though that should be re-thought. The enemy reads as stealthy, mowing down stone walls is the opposite of that.

3. Amalgamations can be spawned indoors. As mentioned in my initial comments, enemies were spawned inside of my base, which made the event much more concerning at the time. I did have a door open, though I was only a tile away from it, but I still don't think that should be happening.

Further, I recently went to the storage area on the top of the hunting lodge, which has barred windows and a metal door, as I needed to disassemble the shelves for pipes, and was caught completely off guard by several swarming amalgams that were inside the room.

To re-iterate, the event spawns enemies inside of closed rooms. I'm fairly certain the curtains were closed as well, which means there was no line of sight either. There need to be checks in place to make sure this does not happen.

4. The event spawns too many self-reviving corpses. If the event lasts 30 in game minutes, the player can be left with up to 90 corpses to track down and dispose of. In my experience I've found amalgamations as far as 3 overmap tiles away from the initial event tile. Having to conduct an exhaustive grid search in a 3+ overmap tile radius to make sure I find the majority of the corpses so that I am not overwhelmed by revives is, in my opinion unreasonable.

This can be solved by copying the amalgamations and giving them the NO_CORPSE flag, as previously mentioned.

5. The event text is unclear. Compare "You shiver as the night closes in…" to "You suddenly register a buzzing in your senses.  It's getting louder, and your head starts to throb.  Somewhere nearby, a tiny cataclysm has begun (check mission log for details)." It is exceedingly clear that something dangerous is about to occur in one of these, whereas the other seems completely innocuous. An event that can randomly spawn on and kill a player should have a little bit more warning to it, in my opinion.

Additionally, the lack of clarity here takes away from the experience. When my first encounter with this event ended, the stand out emotion I felt was confusion. I had no idea what had occurred, I had no ideas as to the mechanics of it, as far as I could tell it was just a random amalgamation storm with some weird narrator watching me from a distance. Only after reading the details on the original pull request did I think it was interesting.

Ultimately I really enjoy the concept of this event, and I think I have a good image of what is intended, but those designs do not currently play out accurately in game.
